block_cascade/cli/crud.py

```diff
@@ -0,0 +1,259 @@
+    if region is None or project is None:
+        gcp_config = get_gcloud_config()
+        if project is None:
+            project = get_gcp_project(gcp_config)
+        if region is None:
+            region = get_gcp_region(gcp_config)
+
+    client_options = {"api_endpoint": f"{region}-{SERVICE}"}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    click.echo("Querying the list of Persistent Resources...")
+    request = aiplatform.ListPersistentResourcesRequest(
+        parent=f"projects/{project}/locations/{region}"
+    )
+
+    response = client.list_persistent_resources(request=request)
+    response_json = json.loads(json_format.MessageToJson(response._pb))
+    click.echo("Listing Persistent Resources:\n")
+    click.echo(json.dumps(response_json, indent=2))
+
+
+@click.command()
+@click.option(
+    "--persistent-resource-id",
+    "-i",
+    required=True,
+    help="ID of the persistent resource to delete.",
+)
+@click.option(
+    "--project",
+    "-p",
+    help="GCP project ID. Inferred from gcloud config if not provided.",
+)
+@click.option(
+    "--region",
+    "-r",
+    help="GCP region. Inferred from gcloud config if not provided.",
+)
+def delete_persistent_resource(
+    persistent_resource_id: str, region: str = None, project: str = None
+):
+    # try to infer project and region from gcloud config
+    if region is None or project is None:
+        gcp_config = get_gcloud_config()
+        if project is None:
+            project = get_gcp_project(gcp_config)
+        if region is None:
+            region = get_gcp_region(gcp_config)
+
+    client_options = {"api_endpoint": get_endpoint_str(region)}
+    client = aiplatform.PersistentResourceServiceClient(client_options=client_options)
+
+    request = aiplatform.DeletePersistentResourceRequest(
+        name=f"projects/{project}/locations/{region}/persistentResources/{persistent_resource_id}"
+    )
+    try:
+        operation = client.delete_persistent_resource(request=request)
+        click.echo("Waiting for operation to complete...")
+        _ = operation.result()
+        click.echo("Persistent resource deleted successfully.")
+    except Exception as e:
+        click.echo(e)
+
+
+@click.command()
+@click.option(
+    "--id",
+    "-i",
+    required=True,
+    help="ID of the persistent resource to describe.",
+)
+@click.option(
+    "--project",
+    "-p",
+    help="GCP project ID. Inferred from gcloud config if not provided.",
+)
+@click.option(
+    "--region",
+    "-r",
+    help="GCP region. Inferred from gcloud config if not provided.",
+)
+def list_active_jobs(id: str, project: str = None, region: str = None):
+    client_options = {"api_endpoint": get_endpoint_str(region)}
+    client = aiplatform.JobServiceClient(client_options=client_options)
+
+    request = aiplatform.ListCustomJobsRequest(
+        parent=f"projects/{project}/locations/{region}",
```

Comment:
Should exit early if project and/or region cannot be read from gcloud configuration.